modules/ROOT/pages/file/file-read.adoc

```diff
@@ -5,36 +5,149 @@ endif::[]
+`MyFiles` in the `resources` folder and a file named `Example.txt` inside of it.
+In Studio, we would see this in our project structure:
+
+image::file/project-resources.png[]
+
+In runtime, the path to access this file would be `${app.home}/MyFiles/Example.txt`.
+To read this file we can first define the following config:
+
+[source,xml,linenums]
+----
+<file:config name="File_Config" >
+    <file:connection workingDir="${app.home}" />
+</file:config>
+----
+
+And then configure the `read` operation like this:
+
+[source,xml,linenums]
+----
+<file:read config-ref="File_Config" path="MyFiles/Example.txt" />
+----
+
+[WARNING]
+As stated before, each time the application is deployed the contents from
+`src/main/resources` folder are copied to the `${app.home}` directory. This means that changes made
+to those files during runtime are lost once the application is stopped or redeployed. This is why we
+recommend putting files in the resource folder that you only need to *read* during your application's
+execution.
+
 == File Locking
 
-You can set a file system lock on the file when it is being read by setting the `lock` parameter to `true` (defaults to `false`). When enabled, this feature asks the operating system to lock the file, which prevents any other process (or Mule flow) from accessing that file while the lock is held. The lock is automatically released when one of the following things occurs:
+You can set a file system lock on the file when it is being read by setting the `lock` parameter
+to `true` (defaults to `false`). When enabled, this feature asks the operating system to lock
+the file, which prevents any other process from accessing that file while the
+lock is held. Because the lock is actually provided by the host file system, its behavior might
+change depending on the mounted drive and the operation system on which mule is running. Take
+that into consideration before blindly relying on this lock. The lock is automatically released
+when one of the following things occurs:
 
 * The Mule flow, which locked the file, ends.
-* The file content has been fully read.
+* The file's content has been fully read.
+
+Note that if the file is already locked by someone else, the connector will not be able to lock
+it, and you will receive a `FILE:FILE_LOCK` error.
+
+An example of locking:
+[source,xml,linenums]
+----
+<file:read config-ref="File_Config" path="filePath" lock="true" />
+----
+
+== Modifying output MimeType and Encoding
+
+If you want to change the MimeType or the encoding of the payload that the `read` operation outputs,
+you can use the `outputEncoding` and `outputMimeType` parameters. For example:
+
+[source,xml,linenums]
+----
+<file:read config-ref="File_Config" path="filePath" outputEncoding="UTF-8" outputMimeType="application/csv; headers=false" />
+----
+
+== Waiting for a file
+
+If you want to read a file, but you are not sure if the file is still being written or not, you can use
```

Comment:
avoid use "you"

Reply:
It's okay to use "you." It's MuleSoft/Salesforce style.